### ISSUE TITLE
Use activity status rather than dates 

### ISSUE THREAD

**Issue by stevieflow** [devinit/D-Portal#674]
As part of d-portal reflecting the IATI standard, it would be good to use directly the ``activity-status`` field to declare the activity status

**Current situation**

Right now, we know that d-portal interprets the ``activity-date`` to then declare activities as either Active or Ended

This makes *logical* sense, but it can disguise poor data quality, which is something we want to now to and avoid, by using d-portal directly

Here's one example (@isabelbirds can supply others) : https://d-portal.org/ctrack.html?reporting_ref=44000#view=act&aid=44000-P127974

![Firefox_Screenshot_2024-04-11T11-20-55 913Z](https://github.com/IATI/D-Portal/assets/1209944/af8b1c59-d90b-4b34-815b-7f7d71516909)

Yes, this has actual-end dates, but the status is _still_ implementation

Right now, this appears in lists of Ended activities:

![Firefox_Screenshot_2024-04-11T11-20-38 913Z](https://github.com/IATI/D-Portal/assets/1209944/52fb4f5d-200b-4671-9dae-2e5bcd495bc8)

**Ideal scenario proposal**

We follow the standard directly and group activities by their [status](https://iatistandard.org/en/iati-standard/203/codelists/activitystatus/):

- 1 	Pipeline/identification 	The activity is being scoped or planned
- 2 	Implementation 	The activity is currently being implemented
- 3 	Finalisation 	Physical activity is complete or the final disbursement has been made, but the activity remains open pending financial sign off or M&E
- 4 	Closed 	Physical activity is complete or the final disbursement has been made.
- 5 	Cancelled 	The activity has been cancelled
- 6 	Suspended 	The activity has been temporarily suspended

I appreciate that is a front-end / layout change ---> but seems more acceptable that us _thinking up_ an grouping mechanism for these status codes, which may in turn disguise things again!

@YohannaLoucheur may be interested in this, given context discussions in https://github.com/IATI/D-Portal/issues/436

**Comment by YohannaLoucheur** [comment]
Agree with your proposal Steven, including keeping the Cancelled and Suspended categories. If the projects are still published, then they should be accessible (if projects are suspended for security reasons, they should be removed from the publisher's file).

We should keep the "Other" category somewhere though - this tends to be a small number of activities with serious data issues (like our 6, with either inverted or missing dates). Seeing this Other category is first intriguing, then a reminder to (try) to resolve the issue.

**Comment by xriss** [comment]
Other is still going to be a needed category for when an invalid code is used or the code is missing.

We currently have 7389 activities missing status codes and one activity that has decided to use a value of 8.

There is also a question regarding the ending soon or starting soon activity lists, these will need rethinking if you want to focus on the published codes rather than dates.

**Comment by stevieflow** [comment]
Thanks @YohannaLoucheur @xriss 

> We currently have 7389 activities missing status codes

I did a quick analysis.  Whilst this will no doubt always be the case, a significant proportion of these activities are from a [single publisher](https://d-portal.org/ctrack.html?reporting_ref=US-501c3-522318905#view=main):

## Activities with no ``activity-status`` ([12 April 2024](https://datasette.codeforiati.org/iati/activity?_sort=rowid&activitystatus_code__isblank=1))
prefix | reportingorg_ref | reportingorg_narrative | activities | COUNTA of iatiidentifier
-- | -- | -- | -- | --
aiddata | US-501c3-522318905 | AidData | 6485 | 86.20%
france |   |   | 531 | 7.06%
  | FR-6 | France, MAEDI | 103 | 1.37%
wwf-uk | GB-COH-4016725 | World Wildlife Fund | 266 | 3.54%
  | GB-COH-1081247 | WWF-UK | 4 | 0.05%
spark | NL-KVK-41213450 | EN: Spark head office Amsterdam | 94 | 1.25%
unrwa | XM-DAC-41130 | United Nations Relief and Works Agency for Palestine Refugees in the Near East | 28 | 0.37%
weeffect | SE-ON-802004-1524 | We Effect (fd Kooperation Utan Gränser) | 2 | 0.03%
frankwater | GB-CHC-1121273 | EN: FRANK Water | 2 | 0.03%
wwf-se | SE-ON-802005-9823 | WWF/World Wide Fund for nature | 1 | 0.01%
kad |   |   | 1 | 0.01%
irishaid |   |   | 1 | 0.01%
fowdk | DK-CVR-78920610 | DA: Verdens Skove, EN: Forests of the World, ES: Bosques del Mundo | 1 | 0.01%
cr |   |   | 1 | 0.01%
buildafrica |   |   | 1 | 0.01%
ai_1064413 |   |   | 1 | 0.01%
acodev | BE-BCE_KBO-0462279234 | FR: ACODEV | 1 | 0.01%
gecfundmanagerpwc | GB-COH-03580586-GEC | Girls' Education Challenge - Fund Manager PwC | 0 | 0.00%
Grand Total |   |   | 7523 | 100.00%

It's unclear what the utility of this publication is, so will investigate separately 

Nevertheless, the "Other" box remains important.  If we are to change the logic to be based on ``activity-status`` then I'd propose that Other is for the non-valid use of this code.  We can work on surfacing date logic errors and other issues (particularly with closer ties to the IATI Validator) as new issues @YohannaLoucheur 

> There is also a question regarding the ending soon or starting soon activity lists, these will need rethinking if you want to focus on the published codes rather than dates.

Thanks @xriss - and appreciate this proposal might start to impact on several long-standing things. I can create a new issue on those lists specifically - it might be useful to know the logic that underpins them, too

Thanks

**Comment by xriss** [comment]
BTW the place where this is a big change is the "active" activities which are about 50% at odds with the published "implementation" status

https://d-portal.org/ctrack.html?status_code=2#view=main

Gives an obvious comparison of the current way vs as published.

**Comment by stevieflow** [comment]
Wow, nice - thanks for highlighting this @xriss 

I still say we proceed!

**Comment by stevieflow** [comment]
@xriss @notshi 

It would be good to resolve this issue.  

- Switching to use ``activity-status`` for the display would mean changing the layout of the boxes (to accommodation other ``activity-status``

> Other is still going to be a needed category for when an invalid code is used or the code is missing.

Agreed

> There is also a question regarding the ending soon or starting soon activity lists, these will need rethinking if you want to focus on the published codes rather than dates.

I think we can continue to use ``activity-date`` for this.  If the boxes with the activity counts have to rejigged, I guess they will not then include any count of "Active activities" or "Ended activities" (as the boxes will reflect the ``activity-status``), so the concept of "Ending soon" will not counter this.

Thanks!

**Comment by stevieflow** [comment]
Hi @notshi @xriss just wanted to check if we have any progress on this one?  Thanks

**Comment by notshi** [comment]
Here are a couple of test pages for the new status section.

1. https://d-portal.org/ctrack.html?test=1#view=main

This is a mixture of the current view with the 6 activity status at the top.

![test1](https://github.com/user-attachments/assets/cdbfd275-1878-4c9d-b93a-5ea2990ab1fc)

2. https://d-portal.org/ctrack.html?test=2#view=main

This is a complete rewrite of the status section, displaying the 6 activity status with total, other and reporting orgs.

This might be the most confusing option for 'first contact' with IATI data (to be honest, it confused us!) as the wordings are not intuitive (friendly) when compared to Active, Planned, Ended, etc.

This is also pushing d-portal to more publisher-focused than user-focused, which d-portal was never intended to be in the first place. The latter something we've always pushed for.

![test2](https://github.com/user-attachments/assets/b396a0ab-5574-4742-ba01-eae25db3a1cd)

**Comment by stevieflow** [comment]
Great, thanks @notshi 

My gut reaction is to continue with option 2, as that is truer to the IATI standard

I do like the layout of the different statuses in the first option, as that looks fairly logical.  

However, I think by *then* displaying activity numbers based on the dates logic continues our pain point - in that it contradicts the statuses

Maybe there is scope whereby we _could_ group statuses into new categories such as "active", but will in turn be an editorial choice we are making. that sits outside of the standard

I think by us using the statuses, and then highlighting to specific publishers cases where their statuses and dates are out of synch, will be a key *data quality* activity, going forward

I fully appreciate your feedback in terms of the departure towards publishers rather than users however 

I'll also speak with @IsabelBirds & @robredpath - and we can discuss

@YohannaLoucheur may also be interested in this

Thanks

-------------

One (minor) think: should the number of Reporting Orgs have a comma separator ?

**Comment by notshi** [comment]
Thanks, @stevieflow - will add a comma separator to Reporting Orgs.

We were wondering if we changed the wording for the current boxes; ie. Active, Planned, Ended, etc, to something more descriptive and included activity start/ends dates, that might be more useful for both users and publishers.

So instead we have these at the top of the standard statuses.
Activities with dates in 2024 (Active)
Activities with end date before 2024 (Ended)
Activities with start date after 2024 (Planned)

![Screenshot_20240905_121841](https://github.com/user-attachments/assets/e74d4887-5117-46eb-952c-b8c787c959fa)

or even this to differentiate the data

![Screenshot_20240905_122417](https://github.com/user-attachments/assets/063ef17b-02ed-4be1-8787-a3bf41482bc8)

**Comment by stevieflow** [comment]
Thank s@shi - really interesting

Certainly agree it's a lot of numbers and boxes!  But- the designs are starting to cement, I can see

The second iteration is really interesting.  

I had assumed that live d-port's use of dates is based on *today* rather than the year we are in - is that not the case?

**Comment by notshi** [comment]
Yes, they are based on *today* but we figured it might be easier to explain and understand with a yearly granularity.

Of course, we will need to adjust the numbers accordingly if we go this route.

**Comment by stevieflow** [comment]
Just discussed with @IsabelBirds 

Important / headline data is : 

- Total activities
- Number of reporters
- Total activities in Implementation

For other statuses, might it be possible to have a plain-ole table?

That leaves the activity-date related dates, which might then sit underneath these headlines

For other activities (activities without a date, or a status?) maybe we need to shaded that differently, or something, to indicate this is a data-quality issue? 

We appreciate that might start to change how we design visualse this part of an activity.  Maybe we can draw some diagrams/sketches or something?

**Comment by stevieflow** [comment]
@notshi having seen this in the IATI colors in https://github.com/IATI/D-Portal/issues/660#issuecomment-2412518679 then it seems fine to go with the layouts shared above

I don't know whether can set the transparency (!) of a box with the design system @robredpath - but that need not block things

**Comment by notshi** [comment]
@stevieflow 
 
> Important / headline data is :
> 
> * Total activities
> * Number of reporters
> * Total activities in Implementation
> 
> For other statuses, might it be possible to have a plain-ole table?
> 

With new theme -
![Screenshot_20250110_152819](https://github.com/user-attachments/assets/4612d9b5-f8e1-4e29-866e-612ccaed100a)

**Comment by stevieflow** [comment]
Nice, thanks @notshi 

Couple of things:

- the use of color is new/interesting - but maybe we revert back to "classic" d-portal and just use the same color for each box ? And/or - red to signify "implementing" might not be the best choice?
- On the list of statuses - I'm not sure why implementation is missing?

Thanks!

**Comment by notshi** [comment]
Sure

- There were very few choices in the design system for button colours so this was mostly trying to use the Secondary and Variant colours. Using the same shade seems to compete with the large header so we might opt for a slightly lighter or darker shade. Out of all the Secondary colours in the design system, red indicates 'active'/ongoing the most.

![Screenshot_20250110_160224](https://github.com/user-attachments/assets/8208c7fa-f5e7-4ca0-b74c-c8318ffa5469)

- Implementation would be repeated if it was listed

**Comment by stevieflow** [comment]
Thanks @notshi 

OK - I see what you mean about the similarity with the header and the box - but I'm not too sure about differently coloured boxes (it's a step too far in my d-portal trauma!)

@robredpath in the design system have we any options or opportunities for a bold color that is not the same as the header blue?

Thanks

**Comment by notshi** [comment]
Have gone for a more subtle approach with the colours but kept the hover states for now.

https://d-portal.org/ctrack.html?&flava=white&test=1#view=main

![Screenshot_20250115_141754](https://github.com/user-attachments/assets/10098f23-fd6a-48e5-8f7b-cc6fb210e1d3)

We have a period of time (usually a week or longer, depending on the nature of changes implemented) for adjustment feedback so changes aren't so reactive.

The header and footer seems as designed to be bold in terms of colour and size of real estate so it would probably benefit the status buttons not to compete in those terms so it feels more balanced.

**Comment by stevieflow** [comment]
Thanks @notshi - looks good / interesting

Regarding this new arrangement of the boxes / lists.  Am I right in assuming this would be on*all* the d-portal themes, at once?  And if so, then we could schedule a Connect post to introduce it (as we did with other things)?

The new theming is secondary to that, in a sense

**Comment by notshi** [comment]
Yes - this will be on all the themes so a Connect post would be good.

https://d-portal.org/ctrack.html?&country_code=PS&/result@type=2&test=1#view=main

![Image](https://github.com/user-attachments/assets/3cb644e2-7ece-4bea-95e8-a5893edfa4b1)

**Comment by stevieflow** [comment]
Thanks @notshi - that's great news

I'll draft something

Note - I'm still not *convinced* we've addressed the whole issue, as we're are mixing IATI standard status codes (which was the original change / intention) with the date-based interpretation (active/ended).  I can really see why  - just wondering aloud

@IsabelBirds any thought?

Thanks

**Comment by stevieflow** [comment]
Note - with the new theme we have some funcky looking boxes

<img width="1659" height="133" alt="Image" src="https://github.com/user-attachments/assets/1312aacc-777f-48db-b362-69795c599c2c" />

**Comment by emmajclegg** [comment]
(from email exchange with @xriss):

> The past / present / future are the same rules as dportal used to use for active / planned / ended before we switched to result codes.
> 
> From the old FAQ:
> - Planned projects are anything with a start date in the future.
> - Active projects are projects with a start date in the past and with an end date in the future.
> - Ended projects are projects with an end date in the past.
> 
> In terms of edge cases, "future" is inclusive of today's date and "actual" dates take precedence over "planned" dates. Note that projects may be open ended, with no actual end date given.
> 
> There could be problems if your PC's time is set wrong. As today's date is sourced from the browser, and not the server, it is possible that it could be set 10 years in the past.
